pkg/controller/certificates/internal/secretsmanager/secret.go

```diff
@@ -205,6 +206,11 @@ func (s *SecretsManager) setValues(crt *cmapi.Certificate, secret *corev1.Secret
 		}
 	}
 
+	// Add DER binary format of the key in the secret
+	if crt.Spec.PrivateKey.WithDerFormat {
+		block, _ := pem.Decode(data.PrivateKey)
+		secret.Data["tls.der"] = block.Bytes
+	}
```

Comment:
Does it make sense for us to supplement the existing key (tls.key), or replace it?

Reply:
Also if a user were to enable this, and then later disable it, the key would continue to exist but just be out of date.

Reply:
I've tried to replace the actual tls.key with the DER formated key, but It failed during the truststore creation because it can't decode it again. That's why I've Just added a boolean option and keep the `tls.key` in PEM format.

Reply:
I can't find a good source on whether `kubernetes.io/tls` secrets _require_ that `tls.crt` and `tls.key` be PEM formatted - [this page](https://kubernetes.io/docs/concepts/configuration/secret/#tls-secrets) mentions that they should be PEM formatted when imported with `kubectl create secret tls ...` and they're certainly PEM formatted when stored after being created in that way.

That said, even if cert-manager didn't assume that `tls.key` is PEM formatted, I would vote that `tls.key` should _always_ exist and should _always_ be PEM formatted, which would imply that we'd add additional `secret.Data` keys (as in this PR) would be the preferred approach if this is something we want to implement.

Reply:
> Also if a user were to enable this, and then later disable it, the key would continue to exist but just be out of date.

NB: I've confirmed this behaviour locally (I didn't doubt it was true but I wanted to check for my own benefit)

Reply:
The JKS and PKCS12 keystores don't have this behaviour, and they delete the keys explicitly:

https://github.com/jetstack/cert-manager/blob/7923823cff0afe5af49282bde3a5a1cc6f5abc84/pkg/controller/certificates/internal/secretsmanager/secret.go#L203

I think this could be replicated for DER keys?